mozilla/activity-stream
# Bug 1542863 - Adding feed load fail fallback.

To test:

1. You need to clone this https://github.com/ScottDowne/fetch-delay locally, run npm install, then node start.js
2. Set your browser.newtabpage.activity-stream.discoverystream.endpoints pref to "https,http"
3. Set your browser.newtabpage.activity-stream.discoverystream.config pref to `{"api_key_pref":"extensions.pocket.oAuthConsumerKey","collapsible":true,"enabled":true,"show_spocs":false,"hardcoded_layout":false,"personalized":false,"adserver":false,"layout_endpoint":"https://gist.githubusercontent.com/ScottDowne/2c5d6774ed644c3f8bcbb00719b8f2b5/raw/f3bb1b513284622d66fd6c4ccdb9b98fb9c3124f/layout-failures" }`
4. load a new tab.

The first feed in this tab should always fail, so you can keep testing the failure state over and over.
Ever feed after that fails on a 20% less likely chance, so the second feed fails 80% of the time, third, 60%, etc.
It may pass on the first fetch. If you want to retry, you need to reset your browser.newtabpage.activity-stream.discoverystream.config off and on again.
